game-core/src/main/java/games/strategy/triplea/ui/mapdata/MapData.java

```diff
@@ -224,52 +216,42 @@ public boolean scrollWrapY() {
     return Boolean.parseBoolean(mapProperties.getProperty(PROPERTY_MAP_SCROLLWRAPY, "false"));
   }
 
-  @Override
-  public void close() {
-    resourceLoader.close();
-  }
-
-  private Map<String, Image> territoryNameImages() {
+  private Map<String, Image> territoryNameImages(final ResourceLoader resourceLoader) {
     if (!resourceLoader.hasPath("territoryNames/")) {
       return new HashMap<>();
     }
 
     final Map<String, Image> territoryNameImages = new HashMap<>();
     for (final String name : centers.keySet()) {
-      final Optional<Image> territoryNameImage = loadTerritoryNameImage(name);
+      final Optional<Image> territoryNameImage = loadTerritoryNameImage(resourceLoader, name);
 
       territoryNameImage.ifPresent(image -> territoryNameImages.put(name, image));
     }
     return territoryNameImages;
   }
 
-  private Optional<Image> loadTerritoryNameImage(final String imageName) {
```

Comment:
The updates to this method are a simplification:
1. We chain the `if( img.isEmpty() )`  check to the optional with an 'or' check
2. Remove exception handling, none of the wrapped code (please double check) throws.

Reply:
Not sure about the exception part.
`loadImage` explicitly rethrows an exception when there's an IOException thrown.
Because we have a URL-null check before, this shoudl happen rarely, but in case the image file has an unsupported format or something, this method would throw an exception.

Reply:
I believe you're right. I added the previous logging to 'loadImage': 8340c9776